R/fit.R

```diff
@@ -365,19 +387,26 @@ mmrm <- function(formula,
     message(paste(fit_msg, collapse = "\n"))
   }
   fit$method <- control$method
-  if (control$method == "Satterthwaite") {
+  fit$cov_method <- control$cov
+  if (identical(control$cov, "Asymptotic")) {
```

Comment:
hm looking at this if/else conditions here we still have very intertwined logic for the method and cov arguments. But I am not sure what is the alternative... e.g. here we check for cov asymptotic, but then we prepare Satterthwaite (method) calculations...